src/mongo_client.ts

```diff
@@ -438,7 +438,11 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> {
         options.ca ??= await fs.readFile(options.tlsCAFile, { encoding: 'utf8' });
       }
       if (typeof options.tlsCertificateKeyFile === 'string') {
-        options.key ??= await fs.readFile(options.tlsCertificateKeyFile, { encoding: 'utf8' });
+        if (typeof options.key !== 'string' || typeof options.cert !== 'string') {
```

Comment:
Are these the right checks? `key` or `cert` could be `Buffer`s as well.

(As a tiny suggestion, I’d maybe even drop the `{ encoding: 'utf8' }` argument here – in theory, this shouldn’t matter much, because the files need to be PEM files and PEM files are ASCII-only, but there’s no real point in decoding the files from UTF-8 because Node.js will just immediately re-encode them using UTF-8, and it might just make the most sense to pass the contents of the file in the most one-to-one fashion possible to Node.js.)

Reply:
In that case would a truthiness check be more appropriate? i.e.

```ts
if (!options.key || !options.cert) {...}
```

I had changed it from this initially since specifying the encoding would make these options of `string` type which would be falsey if the file provided were empty and so would read the file again on a second call to `connect`.

If I'm understanding, dropping the encoding in the `readFile` call would make this work as it would always return a `Buffer` which should always be truthy whether the file is empty or not and so shouldn't read the file again on subsequent `connect` calls.

Reply:
Yeah, as far as I’m concerned a truthiness check sounds right.